general/zstr.hpp

```diff
@@ -203,10 +203,10 @@ class ifstream
    {
       mode |= std::ios_base::in;
       exceptions(std::ios_base::badbit);
-      detail::static_method_holder::check_mode(filename, mode);
+      // detail::static_method_holder::check_mode(filename, mode);
```

Comment:
Is the plan to fix these checks later or should we just remove these?

Reply:
My intention of just commenting the lines was to keep the "diff" to the original zstr source. The fix for the exception is this PR. Regular behavior of an iostream is to only set a failbit.